### PR TITLE
Adds java guide

### PR DESCRIPTION
**Pull request by acadet**
First version of Java style guide.

### CODE REVIEW

File: Java.md [hunk 1]
```diff
@@ -1 +1,201 @@
...
+for (int i = 0; i < list.size(); i++) {
+    // ...
+}
+```
+
+Do not assume size access is constant.
+
+### Variable declarations
+
+Variables should be declared at the top of each block. This process increases performances for it avoids calling memory allocator and garbage collector again.
```
Comment (btaitelb) [inline Java.md]: I thought we discussed this and determined that it doesn't really affect performance in modern JVMs.

File: Java.md [hunk 2]
```diff
@@ -1 +1,201 @@
...
+	
+	// Private & protected methods
+	
+	// Public methods
+}
+```
+
+### Indentation
+
+Standard indentation should be **4 soft spaces**. When a carriage return is needed, keep same indentation.
```
Comment (btaitelb) [inline Java.md]: I always find the term `soft tabs` and `hard tabs` confusing. Can we just say **4 spaces (_no tabs_)** to be clearer?

File: Java.md [hunk 3]
```diff
@@ -1 +1,201 @@
...
+
+Class and method names should be the most limpid as possible. Do not worry about having a long class name. Auto-completion and refactor tools are there for helping you.
+
+Prefer `sortUsersByNameAsc` to `getUsers`.
+
+### Case
+
+In Java world, methods and vars should use **camel case**: `myFantasticMethod` . Class should be **pascal case**: `FriendlyClass`.
+
+Besides, it is a good practice to prefix private and protected methods and fields **by an underscore**. In no time at all, you can easily spot the access of your field/method.
```
Comment (btaitelb) [inline Java.md:50]: This is also good for when we're instrumenting classes for testing, where we have to make a private method public, the underscore still indicates that it's internal.

Comment (btaitelb) [inline Java.md:50]: :+1: to underscores over the prefix _m_ :)

**Comment by btaitelb** [comment]
+1

**Comment by acadet** [comment]
Updated guide

**Comment by btaitelb** [comment]
+1

**Pull request merged**